### PR TITLE
contributing and code_of_conduct

### PR DESCRIPTION
**Pull request by ozgunozerk**
Please take a look, 

I think this can be the first iteration of our contribution guideline.

### CODE REVIEW

**Comment by ozgunozerk** [comment]
We should wait until 0.4.9 becomes stable for merging this PR, because for 0.3.3 we don't have logging.

Also before merging:
- [ ] let Skyler and Diana know about the new Issue Template

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+*For complex implementations you are advised to first discuss the feature implementation or bug fix using an issue.*
+
+
+## Development
+
+### Code style
+
+Subspace is following the
+[Substrate code style](https://github.com/paritytech/substrate/blob/master/docs/STYLE_GUIDE.md).
```
Comment (nazar-pc) [inline CONTRIBUTING.md:58]: No, Subspace is most certainly not following that.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+We have a [Code of Conduct](./CODE_OF_CONDUCT.md), please follow it in all interactions with project maintainers and fellow users :)
+
+## Issues
+
+You want to write an issue if you have a
+- feature request
+- bug report
+- technical question
+
+### How to Create an Issue?
```
Comment (nazar-pc) [inline CONTRIBUTING.md:19]: These should be just GitHub templates if we want to have them at all.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - Submit the log file
+- If it is a feature request:
+    - Clearly explain the feature that is being requested
+    - Explain why do you think we need this feature / how would it enhance the application
+
+**Where is my log file?**
+Below are the `log` file locations (replace `alice` with your username)
+- windows: `C:\Users\Alice\AppData\Roaming\com.subspace.desktop`
+- macos: `/Users/Alice/Library/Logs/com.subspace.desktop`
+- linux: `/home/alice/.config/com.subspace.desktop`
```
Comment (nazar-pc) [inline CONTRIBUTING.md:34]: What are logs doing in `.config` on Linux? They should be in `.local/share` instead. `.config` is for configuration.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+## Development
+
+### Code style
+
+Subspace is following the
+[Substrate code style](https://github.com/paritytech/substrate/blob/master/docs/STYLE_GUIDE.md).
+
+In addition, we incorporate several tools to improve code quality. These are integrated into our CI and are expected to pass before a PR is considered mergeable. They can also be run locally.
+
+* [clippy](https://github.com/rust-lang/rust-clippy) - run with `cargo clippy --release --workspace`
```
Comment (nazar-pc) [inline CONTRIBUTING.md:62]: Is `--release` necessary here? It'll be faster without it.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+The Subspace community is big on decentralization. That is why, it is an open-source project, and are trying our best to be as transparent as possible. To promote Subspace Network being a community driven project, we take any breaches of mutual respect very seriously. Below is the community code of conduct, which applies to our Github repos and organizations, Discord and Telegram channels, and any other Subspace-supported communication group.
```
[the remark this answers is not in the document]
Comment (ozgunozerk) [reply CODE_OF_CONDUCT.md:1]: I wrote this one couple of months ago. I don't exactly recall what I based this on :/
I can do a new research about this.

File: CONTRIBUTING.md (hunk 3 above)
Comment (ozgunozerk) [reply CONTRIBUTING.md:34]: I'm using the default folder choices from `tauri-plugin-log`. Here it is:
https://github.com/tauri-apps/tauri-plugin-log/blob/dev/src/lib.rs#L106

I can open an issue on there to suggest `.local/share` instead.

**Comment by ozgunozerk** [comment]
Turns out there are many things to change in here. And we want our users to use the forum instead of directly creating an issue. 
The main purpose of this PR was to provide a guideline for issue creation. 
I will prepare a github issue template instead. And we can return to this PR later. 
I believe @ImmaZoni is preparing something similar in parallel as well. Maybe he can use the information in here :)

Going to close this one for now. Maybe we can re-open it later

File: CONTRIBUTING.md (hunk 3 above)
Comment (ozgunozerk) [reply CONTRIBUTING.md:34]: Btw, the convention for linux seems to be `var/log`. Is there a reason you prefer `.local/share` instead of `var/log`?

Comment (nazar-pc) [reply CONTRIBUTING.md:34]: Please do, I think it is really important to have logs in the correct location.

No, `/var/log` is for system stuff, regular users don't have access there, and even for system stuff it is generally preferred to go through things like systemd.

**Comment by nazar-pc** [comment]
Yeah, IIRC Justin was preparing something, but can't find it right now. We'd ideally use something like that.

File: CODE_OF_CONDUCT.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ImmaZoni) [reply CODE_OF_CONDUCT.md:1]: > I wrote this one couple of months ago. I don't exactly recall what I based this on :/ I can do a new research about this.

here is the off the shelf one we went with on the subspace docs as was discussed with Nazar a while ago https://github.com/subspace/subspace-docs/blob/main/.github/CODE_OF_CONDUCT.md

[the remark this answers is not in the document]
Comment (nazar-pc) [reply CODE_OF_CONDUCT.md:1]: Yes, I think we want that one everywhere

**Pull request closed without merging**